misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/JBallerinaDebugServer.java

```diff
@@ -655,12 +677,37 @@ private String getStackFrameName(StackFrameProxyImpl stackFrame) {
 
     private Variable[] computeChildVariables(VariablesArguments args) {
         BCompoundVariable parentVar = loadedVariables.get(args.getVariablesReference());
-        Map<String, Value> childVariables = parentVar.getChildVariables();
         Long stackFrameId = variableToStackFrameMap.get(args.getVariablesReference());
         if (stackFrameId == null) {
             return new Variable[0];
         }
-        return childVariables.entrySet().stream().map(entry -> {
+
+        if (parentVar instanceof IndexedCompoundVariable) {
+            // Handles indexed variables.
+            int startIndex = (args.getStart() != null) ? args.getStart().intValue() : 0;
+            int count = (args.getCount() != null) ? args.getCount().intValue() : 0;
+
+            Either<Map<String, Value>, List<Value>> childVars = ((IndexedCompoundVariable) parentVar)
+                    .getIndexedChildVariables(startIndex, count);
+            if (childVars.isLeft()) {
+                // Handles map-type indexed variables.
+                return createVariableArrayFrom(args, childVars.getLeft());
+            } else if (childVars.isRight()) {
+                // Handles list-type indexed variables.
+                return createVariableArrayFrom(args, childVars.getRight());
+            }
+            return new Variable[0];
+        } else if (parentVar instanceof NamedCompoundVariable) {
+            // Handles named variables.
+            Map<String, Value> childVars = ((NamedCompoundVariable) parentVar).getNamedChildVariables();
+            return createVariableArrayFrom(args, childVars);
+        }
+
```

Comment:
```suggestion
```

Reply:
I think we are allowed to keep black lines (maximum 1) within a method to improve readability :slightly_smiling_face: WDYT?